src/mds/MDSContext.h

```diff
@@ -41,8 +41,20 @@ class MDSContext : public Context
  */
 class MDSInternalContextBase : public MDSContext
 {
+  static uint64_t last_seq;
+  uint64_t seq;
```

Comment:
It feels strange to have this in all MDSInternalContext instances, when only some of them will ever need it.

Given that the sorting happens inside MDSCacheObject, maybe the sequence numbers etc should live there too.  What if MDSCacheObject::waiting stored a std::pair of sequence number and MDSCache object for example, and set the sequence number during add_waiter?